zendesk/resource_zendesk_ticket_field.go

```diff
@@ -220,8 +224,6 @@ func resourceZendeskTicketFieldRead(d *schema.ResourceData, meta interface{}) er
 		"editable_in_portal":    field.EditableInPortal,
 		"required_in_portal":    field.RequiredInPortal,
 		"tag":                   field.Tag,
-		"system_field_options":  field.SystemFieldOptions,
```

Comment:
probably should add an issue for this. I think this read method will need to have a bit more logic to transform the api response into the specified schema

Reply:
https://github.com/nukosuke/go-zendesk/blob/master/fixture/GET/ticket_fields.json
This JSON is an actual response which I retrieved from a Zendesk instance. All ticket field types are covered. You should refer to the “type” field to know what additional field should be set.

And this is a workaround for CreateTicketField.
https://github.com/nukosuke/terraform-provider-zendesk/blob/ea3af066714ce25bf95639f84a29473a55bf2204/zendesk/resource_zendesk_ticket_field.go#L166-L177

I hope this helps.

Reply:
take a look now i made some changes based on that json response